drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -231,32 +317,145 @@ static int stm32_clock_control_init(struct device *dev)
+
 #else
-	#error "CONFIG_CLOCK_STM32_PLL_SRC_HSE not selected"
-#endif /* CONFIG_CLOCK_STM32_PLL_SRC_HSE */
+
+	/* No clock source selected for PLL, by default, disable the PLL */
+	LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_NONE);
+	pllsrc_clock = 0;
+#endif
+
+	/* Configure the PLL dividers/multipliers only if PLL source is configured */
+#if defined(CONFIG_CLOCK_STM32_PLL_SRC_HSE) || \
+	defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) || \
+	defined(CONFIG_CLOCK_STM32_PLL_SRC_CSI)
+
+
+	vco_input_range = get_vco_input_range(
+			pllsrc_clock,
+			CONFIG_CLOCK_STM32_PLL_M_DIVISOR);
+
+	__ASSERT(vco_input_range == -ERANGE, "PLL VCO input frequency out of range. Should be from 1 to 16 MHz");
+
+	vco_output_range = get_vco_output_range(vco_input_range);
+
 
 	/* Configure PLL1 */
+	/* According to the RM0433 datasheet */
+	/* Select clock source */
+	/* Init pre divider DIVM */
+	LL_RCC_PLL1_SetM(CONFIG_CLOCK_STM32_PLL_M_DIVISOR);
+	/* Config PLL */
+
+	/* VCO sel, VCO range */
+	LL_RCC_PLL1_SetVCOInputRange(vco_input_range);
+	LL_RCC_PLL1_SetVCOOutputRange(vco_output_range);
+
+	/* FRACN disable DIVP,DIVQ,DIVR enable*/
+	LL_RCC_PLL1FRACN_Disable();
 	LL_RCC_PLL1P_Enable();
 	LL_RCC_PLL1Q_Enable();
 	LL_RCC_PLL1R_Enable();
-	LL_RCC_PLL1FRACN_Disable();
-	LL_RCC_PLL1_SetVCOInputRange(LL_RCC_PLLINPUTRANGE_2_4);
-	LL_RCC_PLL1_SetVCOOutputRange(LL_RCC_PLLVCORANGE_WIDE);
-	LL_RCC_PLL1_SetM(CONFIG_CLOCK_STM32_PLL_M_DIVISOR);
+
+	/* DIVN,DIVP,DIVQ,DIVR div*/
 	LL_RCC_PLL1_SetN(CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER);
 	LL_RCC_PLL1_SetP(CONFIG_CLOCK_STM32_PLL_P_DIVISOR);
 	LL_RCC_PLL1_SetQ(CONFIG_CLOCK_STM32_PLL_Q_DIVISOR);
 	LL_RCC_PLL1_SetR(CONFIG_CLOCK_STM32_PLL_R_DIVISOR);
 
+
+#else
+	/* PLL will stay in reset state configuration */
+#endif /* CONFIG_CLOCK_STM32_PLL_SRC_* */
+
+#if defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL)
+
+	/* Enable PLL*/
 	LL_RCC_PLL1_Enable();
 	while (LL_RCC_PLL1_IsReady() != 1) {
 	}
 
+
+	/* Set PLL1 as System Clock Source */
+	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_PLL1);
+	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_PLL1) {
+	}
+
+#elif defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_HSE)
+
+	/* Enable HSI oscillator */
+	LL_RCC_HSE_Enable();
+	while (LL_RCC_HSE_IsReady() != 1) {
+	}
+
+	/* Set sysclk source to HSE */
+	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_HSE);
+	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_HSE) {
+	}
+
+#elif defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_HSI)
+	/* Check if we want HSI as clock source instead of PLL */
+	/* If HSI is main clock source and PLL clock source */
+	/* then PLL cannot clock USB and RC48 must be used for USB */
+
+	/* Enable HSI oscillator */
+	LL_RCC_HSI_Enable();
+	while (LL_RCC_HSI_IsReady() != 1) {
+	}
+
+	/* Set sysclk source to HSI */
+	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_HSI);
+	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_HSI) {
+	}
+
+#elif defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_CSI)
+
+	/* Enable HSI oscillator */
```

Comment:
```suggestion
	/* Enable CSI oscillator */
```

Reply:
Done, corrected for this comment and the following one.